File: Angula2BaseStructure/Common/Entities/Dorm/Dorm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Dorm
{
    public class Dorm
    {
        public int DormId { get; set; }
        public string Name { get; set; }
        public virtual List<Room> Rooms { get; set; }
        public virtual List<Room> RoomsGroundFloor { get; set; }
        public virtual List<Room> RoomsFloor1 { get; set; }
        public virtual List<Room> RoomsFloor2 { get; set; }
        public virtual List<Room> RoomsFloor3 { get; set; }
        public virtual List<Room> RoomsFloor4 { get; set; }
        public virtual List<Room> RoomsFloor5 { get; set; }
        public List<int> RestrictedRoomNames { get; set; }
    }
}

File: Angula2BaseStructure/Angula2BaseStructure/Scripts/App/Entities/Dorm/Dorm.ts
import {Room} from './Room';
export class Dorm {
    dormId: number;
    name: string;
    rooms: Array<Room>;
    roomsGroundFloor: Array<Room>;
    roomsFloor1: Array<Room>;
    roomsFloor2: Array<Room>;
    roomsFloor3: Array<Room>;
    roomsFloor4: Array<Room>;
    roomsFloor5: Array<Room>;
    restrictedRoomNames = [];


}